cli/tests/unit/timers_test.ts

```diff
@@ -366,3 +366,33 @@ unitTest(async function timerNestedMicrotaskOrdering(): Promise<void> {
 unitTest(function testQueueMicrotask() {
   assertEquals(typeof queueMicrotask, "function");
 });
+
+unitTest(async function testTimerIgnoresDateOverride(): Promise<void> {
+  const OriginalDate = Date;
+  const { promise, resolve, reject } = deferred();
+  let hasThrown = 0;
+  try {
+    globalThis.Date = {
+      now: () => {
+        reject("global Date override used over original Date object");
+        return 0;
+      },
+    } as DateConstructor;
+    setTimeout(() => resolve(), 1);
+    await promise;
+    hasThrown = 1;
+  } catch (err) {
+    if (typeof err === "string") {
+      globalThis.Date = OriginalDate;
```

Comment:
This is unnecessary since it's done in the finally block?

Reply:
assertEquals uses global Date object, if it's not restored it will throw an error because Date is overridden with a non function value. That's why I restore it in this specific error case. This error case would only happen if someone changes timers to use the global Date again. With the change I made, no exception is thrown. I could change this to instead restore it at the end of the try block and start of the catch block, then remove the finally block if that would be preferred.

Reply:
Actually, it might be better if I override Date with a function, then add on all the static methods, having it reject the promise if any Date function is called. That way it can catch any use of the Date override and can just have restoration done in the finally block.

Reply:
> Actually, it might be better if I override Date with a function, then add on all the static methods, having it reject the promise if any Date function is called. That way it can catch any use of the Date override and can just have restoration done in the finally block.

Should we wait for this, or land the PR as-is?